### PR TITLE
Validate arrays of objects

### PR DESCRIPTION
**Pull request by jsoriano**
Add validation for arrays of objects, considering them as valid when they are inside `nested` or `group`.

When `group` is used, a filterable error is given, mentioning that `nested` should be used instead on this case.

Fix https://github.com/elastic/elastic-package/issues/1488

### CODE REVIEW

File: internal/fields/validate.go [hunk 1]
```diff
@@ -27,13 +27,20 @@ import (
 	"github.com/elastic/elastic-package/internal/multierror"
 	"github.com/elastic/elastic-package/internal/packages"
 	"github.com/elastic/elastic-package/internal/packages/buildmanifest"
+	"github.com/elastic/package-spec/v2/code/go/pkg/specerrors"
 )
 
+// EPF - Elastic Package Fields [validation]
+const ArrayOfObjectsErrorCode = "EPF00001"
```
Comment (jsoriano) [inline internal/fields/validate.go:34]: @mrodm do we have a place where we are already defining Elastic Package errors?

Comment (mrodm) [reply internal/fields/validate.go:35]: Currently, there is no definition of errors in Elastic Package. Until now, it was just getting the errors from package-spec validate method and return the errors filtered (if any).

Should this be added into a new `internal/errors` package ? Or maybe another naming to not have collisions with standard errors package (e.g. validationerrors? I don't like neither too much this).

File: internal/fields/validate.go [hunk 2]
```diff
@@ -783,11 +791,22 @@ func (v *Validator) parseSingleElementValue(key string, definition FieldDefiniti
...
 		case map[string]interface{}:
-			// TODO: This is probably an element from an array of objects,
+			// This is probably an element from an array of objects,
 			// even if not recommended, it should be validated.
+			if v.specVersion.LessThan(semver3_0_0) {
+				break
+			}
+			errs := v.validateMapElement(key, common.MapStr(val), doc)
+			if definition.Type == "group" {
+				errs = append(errs, arrayOfObjectsErr)
```
Comment (mrodm) [inline internal/fields/validate.go:804]: I don't think these errors could be filtered as it is the code.

The methods that get the errors and filter are just for the errors from package-spec. This is in `internal/validation` package. For instance
https://github.com/elastic/elastic-package/blob/29ded9d537b0fc2f9f62b66859cbb607726741fc/internal/validation/validation.go#L26

File: internal/fields/validate.go (hunk 1 above)
Comment (jsoriano) [reply internal/fields/validate.go:35]: Removed by now.

File: internal/fields/validate.go (hunk 2 above)
Comment (jsoriano) [reply internal/fields/validate.go:804]: Removed by now.

**Review by mrodm — APPROVED** [review]

**Pull request merged**